Sources/TokamakDOM/Views/Text.swift

```diff
@@ -17,9 +17,106 @@ import TokamakCore
 
 public typealias Text = TokamakCore.Text
 
+extension Font.Design: CustomStringConvertible {
+  /// Some default font stacks for the various designs
+  public var description: String {
+    switch self {
+    case .default:
+      return "system, -apple-system, \".SFNSText-Regular\", \"San Francisco\", \"Roboto\", \"Segoe UI\", \"Helvetica Neue\", \"Lucida Grande\", sans-serif"
+    case .monospaced:
+      return "Consolas, \"Andale Mono WT\", \"Andale Mono\", \"Lucida Console\", \"Lucida Sans Typewriter\", \"DejaVu Sans Mono\", \"Bitstream Vera Sans Mono\", \"Liberation Mono\", \"Nimbus Mono L\", Monaco, \"Courier New\", Courier, monospace"
+    case .rounded: // Not supported due to browsers not having a rounded font builtin
+      return Self.default.description
+    case .serif:
+      return "Cambria, \"Hoefler Text\", Utopia, \"Liberation Serif\", \"Nimbus Roman No9 L Regular\", Times, \"Times New Roman\", serif"
+    }
+  }
+}
+
+extension Font.Leading: CustomStringConvertible {
+  public var description: String {
+    switch self {
+    case .standard:
+      return "normal"
+    case .loose:
+      return "1.5"
+    case .tight:
+      return "0.5"
+    }
+  }
+}
+
+extension Font: StylesConvertible {
+  var styles: [String: String] {
+    [
+      "font-family": _name == _FontNames.system.rawValue ? _design.description : _name,
+      "font-weight": "\(_bold ? 700 : _weight.value)",
+      "font-style": _italic ? "italic" : "normal",
+      "font-size": "\(_size)",
+      "line-height": _leading.description,
+      "font-variant": _smallCaps ? "small-caps" : "normal",
+    ]
+  }
+}
+
 extension Text: AnyHTML {
   public var innerHTML: String? { textContent(self) }
   var tag: String { "span" }
-  var attributes: [String: String] { [:] }
+  var attributes: [String: String] {
+    var font: Font?
+    var color: Color?
+    var italic: Bool = false
+    var weight: Font.Weight?
+    var kerning: String = "normal"
+    var baseline: CGFloat?
+    var strikethrough: (Bool, Color?)?
+    var underline: (Bool, Color?)?
+    for modifier in _modifiers {
+      switch modifier {
+      case let .color(_color):
+        color = _color
+      case let .font(_font):
+        font = _font
+      case .italic:
+        italic = true
+      case let .weight(_weight):
+        weight = _weight
+      case let .kerning(_kerning), let .tracking(_kerning):
+        kerning = "\(_kerning)em"
+      case let .baseline(_baseline):
+        baseline = _baseline
+      case .rounded: break
+      case let .strikethrough(active, color):
+        strikethrough = (active, color)
+      case let .underline(active, color):
+        underline = (active, color)
+      }
+    }
+    let hasStrikethrough = strikethrough?.0 ?? false
+    let hasUnderline = underline?.0 ?? false
+    let textDecoration = "\(!hasStrikethrough && !hasUnderline ? "none" : "")\(hasStrikethrough ? "line-through" : "") \(hasUnderline ? "underline" : "")"
```

Comment:
**chore, nitpick:** Long line. I realise that formatting this manually is tedious, my main motivation here is to make this code more readable on different devices without wide screens so that horizontal scrolling is not required, such as laptops, tablets and phones. 🙂 I hope that SwiftFormat gains support for automatic splitting of long lines at some point 🙏